compiler/src/dmd/common/md5.d

```diff
@@ -7,7 +7,7 @@
  **********************************************************************
  */
 
-module dmd.backend.md5;
+module dmd.common.md5;
```

Comment:
I've not seen this yet being used by any frontend module - e2ir and toobj are not really frontend - to warrant this move though.

One slight concern I have is that the licence of this module doesn't seem to be compatible with gdc, but that will only be a (small) problem if this starts to be imported by any language parser/semantic modules - it'll probably be no issue to  supply a libre version of the module.